### PR TITLE
Avoid conversion to blob when responseText used.

### PR DESCRIPTION
**Pull request by lrowe**
String decoding / encoding is likely to render the binary data corrupt so throw error instead.

### CODE REVIEW

File: fetch.js [hunk 1]
```diff
@@ -106,6 +106,9 @@
 
     if (blobSupport) {
       this.blob = function() {
+        if (!this._bodyBlob) {
+          throw new Error('XMLHttpRequest missing blob support')
+        }
```
Comment (josh) [inline fetch.js]: I don't think this is correct. A `Response` can be constructed with a String and accessed from the blob mixing.

**Comment by josh** [comment]
For response/body's `bodyInit`, its recommend that the value be read into a stream first.

https://fetch.spec.whatwg.org/#concept-bodyinit-extract

Eventually when whatwg streams are furthered spec'd they'll be decoded into the string instead of the blob.

**Comment by josh** [comment]
Added some additional tests to master to show you what I mean 9d586486e50a79551b1d12178b3408d1fd57cb35

**Comment by lrowe** [comment]
This is #61 again. IE 8/9 in combination with a Blob polyfill will corrupt binary data. It seems better to disallow usage of blob/arrayBuffer in that case.

To make the `Response` constructor compatible maybe the xhr.responseText could be wrapped in a special object to indicate that returning a blob is disallowed?

**Comment by josh** [comment]
What blob polyfill are you using?

**Comment by lrowe** [comment]
I'm not using a blob polyfill myself, though from #65 it looks like @kruppel is. I'd just like to rule out any possibility of data corruption.

**Comment by josh** [comment]
If the polyfill Blob constructor doesn't work, I'm not sure what good it is.

**Comment by lrowe** [comment]
I guess it could be useful for processing binary data in other ways, but it does not fix XMLHttpRequest for you. For that you would need the ugly VBScript work around.

**Comment by lrowe** [comment]
Probably worth noting somewhere that with XMLHttpRequest 1 or without Blob support the charset will be auto-detected from the content-type header whereas spec is to always use utf8. So there's an argument for disabling `text()` on IE8/9 too, but that's probably going too far ;)

File: fetch.js [hunk 2]
```diff
@@ -228,7 +229,9 @@
       options = {}
     }
 
-    if (blobSupport) {
+    if (bodyInit instanceof XMLHttpRequest) {
```
Comment (dgraham) [inline fetch.js]: This is non-standard behavior.

**Comment by dgraham** [comment]
This branch is speculative—let's deal with errors as they arise. Blob usage is uncommon right now.

**Pull request closed without merging**